### PR TITLE
Add more details on repository configuration to documentation

### PR DESCRIPTION
**Pull request by alexhung**
Thanks for suggestion from @oallauddin https://github.com/jfrog/terraform-provider-project/issues/58#issuecomment-1319445493

You can preview the content by copy/paste into Terraform's [Doc Preview tool](https://registry.terraform.io/tools/doc-preview).

### CODE REVIEW

**Review by MahithaB — APPROVED** [review]

File: docs/resources/project.md [hunk 1]
```diff
@@ -4,14 +4,45 @@ page_title: "project Resource - terraform-provider-project"
...
+  lifecycle {
+      ignore_changes = [
+          project_environments,
+          project_key
+      ]
+  }
+  
+  }
+  ``
+  ~>We strongly recommend using therepos` attribute to manage the list of repositories. See below for additional details.
```
Comment (danielmkn) [inline docs/resources/project.md:22]: therepos` attribute ?

File: docs/resources/project.md [hunk 2]
```diff
@@ -4,14 +4,45 @@ page_title: "project Resource - terraform-provider-project"
 subcategory: ""
 description: |-
   Provides an Artifactory project resource. This can be used to create and manage Artifactory project, maintain users/groups/roles/repos.
-  ~>We strongly recommend using the "repos" attribute to manage the list of repositories. See below for more details.
+  Repository Configuration
+  After the project configuration is applied, the repository's attributes project_key and project_environments would be updated with the project's data. This will generate a state drift in the next Terraform plan/apply for the repository resource. To avoid this, apply lifecycle.ignore_changes:
```
Comment (danielmkn) [inline docs/resources/project.md:8]: it's odd that doc plugin removed quotes here: `attributes project_key and project_environments would`

**Review by danielmkn — CHANGES_REQUESTED** [review]
Please fix typo.

File: docs/resources/project.md (hunk 2 above)
Comment (alexhung) [reply docs/resources/project.md:8]: Yeah, since these are in the YAML Frontmatter section they don't affect the actual documentation.

**Review by shrajfr12 — APPROVED** [review]

**Review by danielmkn — APPROVED** [review]

**Pull request merged**